In [11]:
#import libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

import warnings
warnings.filterwarnings("ignore")

In [12]:
#read the dataset
filename = pd.read_csv("StudentsPerformance.csv")
df = pd.DataFrame(filename)

In [13]:
df.head()

gender race/ethnicity parental level of education         lunch  \
0  female        group C            some high school  free/reduced   
1  female        group B                 high school  free/reduced   
2  female        group B            some high school  free/reduced   
3  female        group B                some college      standard   
4  female        group C                some college  free/reduced   

  test preparation course  math score  reading score  writing score  
0                    none           0             17             10  
1                    none           8             24             23  
2                    none          18             32             28  
3                    none          11             38             32  
4                    none          22             39             33

In [14]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 8 columns):
 #   Column                       Non-Null Count  Dtype 
---  ------                       --------------  ----- 
 0   gender                       1000 non-null   object
 1   race/ethnicity               1000 non-null   object
 2   parental level of education  1000 non-null   object
 3   lunch                        1000 non-null   object
 4   test preparation course      1000 non-null   object
 5   math score                   1000 non-null   int64 
 6   reading score                1000 non-null   int64 
 7   writing score                1000 non-null   int64 
dtypes: int64(3), object(5)
memory usage: 62.6+ KB


# 1. Find out how many males and females participated in the test?

In [15]:
df['gender'].value_counts()

female    518
male      482
Name: gender, dtype: int64

# 2. What do you think about the students parental level of education?

In [16]:
df['parental level of education'].value_counts()

some college          226
associate's degree    222
high school           196
some high school      179
bachelor's degree     118
master's degree        59
Name: parental level of education, dtype: int64

Most of them are having some college level of education. The number of parents having some college level education and having associate's degree are almost equal. Only a few have master's degree.

# 3. Who scores the most on average for math, reading and writing based on

# ● Gender

In [17]:
df1=df.groupby('gender')['math score'].mean()
df1

gender
female    63.376448
male      68.821577
Name: math score, dtype: float64

On average for math, males scores the most than the females.

In [18]:
df2=df.groupby('gender')['reading score'].mean()
df2

gender
female    72.590734
male      65.545643
Name: reading score, dtype: float64

On average for reading, females scores the most than the males.

In [19]:
df3=df.groupby('gender')['writing score'].mean()
df3

gender
female    72.467181
male      63.446058
Name: writing score, dtype: float64

On average for writing, females scores the most than males.

# ● Test preparation course

In [20]:
df4=df.groupby('test preparation course')['math score'].mean()
df4

test preparation course
completed    69.96648
none         63.78972
Name: math score, dtype: float64

On average for math, those students whose test preparation course was completed scores the most.

In [21]:
df5=df.groupby('test preparation course')['reading score'].mean()
df5

test preparation course
completed    74.175978
none         66.417445
Name: reading score, dtype: float64

On average for reading, those students whose test preparation course was completed scores the most.

In [22]:
df6=df.groupby('test preparation course')['writing score'].mean()
df6

test preparation course
completed    74.684358
none         64.457944
Name: writing score, dtype: float64

On average for writing, those students whose test preparation course was completed scores the most.

# 4. What do you think about the scoring variation for math, reading and writing based on

# ● Gender

In [23]:
df1=df.groupby('gender')['math score'].std()
df1

gender
female    16.029928
male      14.556411
Name: math score, dtype: float64

In [24]:
df2=df.groupby('gender')['reading score'].std()
df2

gender
female    14.411018
male      14.149594
Name: reading score, dtype: float64

In [25]:
df3=df.groupby('gender')['writing score'].std()
df3

gender
female    14.844842
male      14.227225
Name: writing score, dtype: float64

Males are more consistent in scoring in math,reading and writing than females.

# ● Test preparation course

In [26]:
df4=df.groupby('test preparation course')['math score'].std()
df4

test preparation course
completed    14.521847
none         15.705689
Name: math score, dtype: float64

In [27]:
df5=df.groupby('test preparation course')['reading score'].std()
df5

test preparation course
completed    13.537572
none         14.608896
Name: reading score, dtype: float64

In [28]:
df6=df.groupby('test preparation course')['writing score'].std()
df6

test preparation course
completed    13.236412
none         15.041667
Name: writing score, dtype: float64

The students whose test preparation course was completed scored in all fields more consistently.

# 5. The management needs your help to give bonus points to the top 25% of students based on their maths score, so how will you help the management to achieve this?

In [52]:
Q1=np.quantile(df['math score'],0.75)
Q1

77.0

In [59]:
np.percentile(df['math score'],[25,50,75])

array([57., 66., 77.])

In [60]:
df.describe()

math score  reading score  writing score
count  1000.000000    1000.000000    1000.000000
mean     66.001000      69.195000      68.119000
std      15.569567      14.706436      15.226381
min       0.000000      17.000000      10.000000
25%      57.000000      59.000000      58.000000
50%      66.000000      70.000000      69.000000
75%      77.000000      79.250000      79.000000
max     100.000000     100.000000     100.000000

In [61]:
df[df['math score']>77]

gender race/ethnicity parental level of education         lunch  \
430  female        group A                some college      standard   
431  female        group B           bachelor's degree  free/reduced   
432  female        group D           bachelor's degree      standard   
433  female        group D                 high school      standard   
434  female        group D           bachelor's degree  free/reduced   
..      ...            ...                         ...           ...   
995    male        group E                some college      standard   
996    male        group A                some college      standard   
997    male        group D                some college      standard   
998    male        group E          associate's degree  free/reduced   
999    male        group E           bachelor's degree      standard   

    test preparation course  math score  reading score  writing score  
430               completed          78             87             91  
431                    none          78             79             76  
432                    none          78             82             79  
433                    none          78             81             80  
434                    none          78             90             93  
..                      ...         ...            ...            ...  
995               completed          99             87             81  
996               completed         100             96             86  
997               completed         100             97             99  
998               completed         100            100             93  
999               completed         100            100            100  

[231 rows x 8 columns]

The students who scored more than 77 in math comes under top 25%. There are 231 students who scored more than 77 in math.